medicines/search-client/src/lib.rs

```diff
@@ -166,43 +192,70 @@ impl CreateIndexEntry for AzureSearchClient {
 
 async fn search(
     search_term: &str,
+    result_count: Option<u32>,
+    offset: Option<u32>,
+    include_count: Option<bool>,
     client: &reqwest::Client,
     config: &AzureConfig,
-) -> Result<IndexResults, reqwest::Error> {
-    let req = build_search(search_term, &client, &config)?;
+) -> Result<IndexResults, anyhow::Error> {
+    let req = build_search(
+        search_term,
+        result_count,
+        offset,
+        include_count,
+        &client,
+        &config,
+    )?;
     tracing::debug!("Requesting from URL: {}", &req.url());
     client
         .execute(req)
         .await?
         .error_for_status()?
         .json::<IndexResults>()
         .await
+        .map_err(|e| anyhow::anyhow!(e))
 }
 
 fn build_search(
     search_term: &str,
+    result_count: Option<u32>,
+    offset: Option<u32>,
```

Comment:
could these be `i32` instead? I don't know what a negative integer would be useful for

Reply:
They could be `i32`, but that would allow negative result count and offsets, which don't make any sense.

Reply:
right, think I just muddled up which integer is which, don't mind me